### DISCUSSION TITLE
Optional Path parameter

### DISCUSSION THREAD

**Discussion by qarmin** [actix/actix-web#2874] [Q&A] [answered]
When I tried to use this route, then actix properly returned code 200 for get `http://127.0.0.1:8081/patient-info/`
```rust
#[get("/patient-info/")]
pub async fn AAAA(
    pool: web::Data<DbPool>,
) -> anyhow::Result<HttpResponse, Error> {
    Ok(HttpResponse::Ok().body("ABCD"))
}
```
but when I used this one, then same get returned 404
```rust
#[get("/patient-info/{eye}")]
pub async fn AAAA(
    pool: web::Data<DbPool>,
    path: web::Json<Option<String>>,
) -> anyhow::Result<HttpResponse, Error> {
    Ok(HttpResponse::Ok().body("ABCD"))
}
```
How can I get eye value(can be none if request is `patient-info/` or not none when using `patient-info/LEFT`)?

**Comment by robjtede** [comment, marked as answer]
```rust
#[get("/patient-info/{eye}")]
pub async fn AAAA(
    pool: web::Data<DbPool>,
    path: web::Path<String>,
) -> anyhow::Result<HttpResponse, Error> {
    Ok(HttpResponse::Ok().body("ABCD"))
}
```

Though `Option<String>` will not work here. You'll need two handlers and a delegate method. Eg:

```rust
#[get("/patient-info")]
pub async fn aaa(
    pool: web::Data<DbPool>,
    path: web::Path<String>,
) -> Result<HttpResponse, Error> {
  aaaa_inner(&pool, None)
}

#[get("/patient-info/{eye}")]
pub async fn aaaa_eye(
    pool: web::Data<DbPool>,
    path: web::Path<String>,
) -> Result<HttpResponse, Error> {
  aaaa_inner(&pool, Some(path.into_inner()))
}

async fn aaaa_inner(pool: &DbPool, path: Option<String>) -> Result<HttpResponse, Error> {
  ...
}
```

**Reply by happybeing** [reply]
I have the same question, but not just how to implement the functionality. I am using utoipa's swagger support, but guess there's no way to have this appear as a single route with optional path parameter in the SwaggerUI. 

Having two similar routes in the docs and SwaggerUI is I guess the only option here.

**Reply by robjtede** [reply]
OpenAPI (previously swagger) doesn't actually allow optional path parameters. These are semantically two different resource paths.
